### PR TITLE
[Fixes #7212, Fixes #7329] Orchard.Comments - Rework frontend and fix bugs

### LINKED ISSUE #7212: Can't have two Content Items with Threaded Comments Part on one page
Having two Content Items with Comments Part on a single page works fine, but if they are both threaded, the threaded feature breaks--it makes a comment as if it were not threaded. 

Also, when one hits "reply" on either of the two Content Item Comment Parts, the second Comment Part drops out of its own zone and joins the first Comment Part in its zone.

### LINKED ISSUE #7329: Orchard.Themes - TheThemeMachine - Fix textarea overflow issue
Stop the textarea overflowing when its in a narrow column (eg comments in sidebar):

image

I already have the fix, this is just a placeholder, will submit the PR after finishing with current work.

```
.comment-form .comment-text {
    max-width: 33.2em;
    width: 97%;
}
```

### PR DESCRIPTION
**Pull request by rtpHarry**
[Fixes #7212, Fixes #7329] Orchard.Comments - Rework frontend and fix bugs
## Changelog
- BUG - comment form did not work if another content type with a CommentsPart was on the same page, #7212
- BUG - a comment with many replies was difficult to use as the reply button was at the top but the reply form appeared below, after all of its children. Moved the reply and the comment button down to end of comment text - natural location, the form is always next to it then when reply button is pressed.
- BUG - clicking reply, then clicking reply on a second item would return the comment form to its home location at the end of the page rather than moving it to its newly requested place
- BUG - fixed comment form overflow, #7329
- BUG - removed css rules in TheThemeMachine referencing id's and non-existent elements
- BUG - fixed w3 validator errors - Bad value 2016-10-16T20:22:28.0000000Z for attribute datetime on element time: A fraction of a second must be one, two, or three digits.
- BUG - removed all but one of the duplicate ids in the comment forms
- BUG - added missing dependency on Orchard.Resources in module.txt
- IMPROVEMENT - added permalink to comment from admin panel
- IMPROVEMENT - decoupled comment form insertion from template. Now the script looks for a .comment-reply-form-beacon per comment rather than relying on jquery combo parent() and searching for article tags
- IMPROVEMENT - added animation - reply form slides into view
- IMPROVEMENT - moved script to resourcemanifest so its not duplicated and can be cached
- IMPROVEMENT - added assets.json pipeline support for automatic minification

### CODE REVIEW

**Comment by rtpHarry** [comment]
## Breaking changes

  - All CSS ids have been removed from the views
  - Reply button has moved from top row of comment to under the comment text
  - JavaScript has been rewritten to use data attributes instead of ids

## Theme authors

  - Move the div with `.comment-reply-form-beacon` to the place you want the comment form to be dynamically injected.

## Issue - Generating unique ids

I looked into it and it seems that it basically goes against the intention of MVC model binding to have more than one thing on the page so I don't know if there is a built in way to do this.

If there are three content types on the page then there are three duplicate ids for each form.

I have changed the ids with code like this:

    @Html.HiddenFor(m => m.RepliedOn, new {
        @class = "comments-repliedon",
        data_contentitem_id = Model.CommentedOn,
        id = string.Format("Comments_RepliedOn_{0}", Model.CommentedOn)
    })

This works but then the labels `for=` are all pointing at the wrong fields so I had to set the ids for them as well.

Manually handling all the form inputs and label ids seems like a poor approach. I wondered if anyone knows a way to inject an extra piece of data (like the containerid) into the pipeline so it will automatically generate unique sets of ids?

## Issue - Remaining duplicate id

A remaining part is being output with empty values so it seems useless:

    <input id="CommonPart_ContainerId" name="CommonPart.ContainerId" type="hidden" value="">

Shape Tracer didn't identify it correctly. Is there a way I can hide it or can I edit the ID like the rest via the code?

## Issue - Breaks if the same content item is on the page twice

For example in my testing, I displayed the comments form on the blog summary. With a RecentBlogPosts widget in the sidebar this results in

**Comment by rtpHarry** [comment]
## Testing
As this is a big merge you may want to test it by checking out the code. I would suggest going to the `Placement.info` in Orchard.Comments and replacing the Summary match with this:

    <Match DisplayType="Summary">
        <Place Parts_Comments_Count="Meta:5" />
        <Place Parts_ListOfComments="Content:10.2" />
        <Place Parts_CommentForm="Content:10.3" />
    </Match>

That will show the comments and the form after each post in the main index of the blog.

Here is a recipe that will populate it with some random posts and comments:

      <?xml version="1.0" encoding="utf-8" standalone="yes"?>
      
      <Orchard>
        <Recipe>
          <ExportUtc>2016-10-17T22:45:08.6208632Z</ExportUtc>
        </Recipe>
        <Content>
          <Blog Id="/alias=" Status="Published">
            <AdminMenuPart AdminMenuText="" OnAdminMenu="false" />
            <AutoroutePart Alias="" CustomPattern="/" UseCustomPattern="true" UseCulturePattern="false" PromoteToHomePage="false" />
            <BlogPart Description="This is your Orchard Blog." PostCount="4" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-15T19:39:02.9514779Z" PublishedUtc="2016-10-15T19:39:02.9694907Z" ModifiedUtc="2016-10-15T19:39:02.9514779Z" />
            <TitlePart Title="Blog" />
          </Blog>
          <BlogPost Id="/alias=example-with-featured-image" Status="Published">
            <AutoroutePart Alias="example-with-featured-image" UseCustomPattern="false" UseCulturePattern="false" PromoteToHomePage="false" />
            <BodyPart Text="&lt;p&gt;&amp;lt;Place Parts_ListOfComments=&quot;Content:10.2&quot; /&amp;gt;&lt;br /&gt; &amp;lt;Place Parts_CommentForm=&quot;Content:10.3&quot; /&amp;gt;&lt;/p&gt;" />
            <CommentsPart CommentsShown="true" CommentsActive="true" ThreadedComments="true" />
            <CommonPart Owner="/User.UserName=admin" Container="/alias=" CreatedUtc="2016-10-15T22:34:49.4416279Z" PublishedUtc="2016-10-15T22:34:49.4416279Z" ModifiedUtc="2016-10-15T22:34:49.4321207Z" />
            <TagsPart Tags="" />
            <TitlePart Title="Example With Featured Image" />
          </BlogPost>
          <BlogPost Id="/alias=single-thread-jed" Status="Published">
            <AutoroutePart Alias="single-thread-jed" UseCustomPattern="false" UseCulturePattern="false" PromoteToHomePage="false" />
            <BodyPart Text="&lt;p&gt;&lt;span&gt;If you've run the default setup, you already noticed that Nitrate can handle IIS Express. Note however that right now this does not support HTTPS. If you need to work with SSL in Orchard, you should configure and run IIS Express from within Visual Studio instead.&lt;/span&gt;&lt;/p&gt;" />
            <CommentsPart CommentsShown="true" CommentsActive="true" ThreadedComments="false" />
            <CommonPart Owner="/User.UserName=admin" Container="/alias=" CreatedUtc="2016-10-17T21:27:15.4220118Z" PublishedUtc="2016-10-17T21:27:15.4220118Z" ModifiedUtc="2016-10-17T21:27:15.340454Z" />
            <TagsPart Tags="" />
            <TitlePart Title="Single Thread Jed" />
          </BlogPost>
          <BlogPost Id="/alias=test-post-1" Status="Published">
            <AutoroutePart Alias="test-post-1" UseCustomPattern="false" UseCulturePattern="false" PromoteToHomePage="false" />
            <BodyPart Text="&lt;p&gt;l;khklhjkh kjh kjh kjh kjh iug iuh k&lt;/p&gt;" />
            <CommentsPart CommentsShown="true" CommentsActive="true" ThreadedComments="true" />
            <CommonPart Owner="/User.UserName=admin" Container="/alias=" CreatedUtc="2016-10-15T22:31:50.6560719Z" PublishedUtc="2016-10-15T22:31:50.6560719Z" ModifiedUtc="2016-10-15T22:31:50.537988Z" />
            <TagsPart Tags="" />
            <TitlePart Title="Test Post 1" />
          </BlogPost>
          <BlogPost Id="/alias=third-post" Status="Published">
            <AutoroutePart Alias="third-post" UseCustomPattern="false" UseCulturePattern="false" PromoteToHomePage="false" />
            <BodyPart Text="&lt;p&gt;InitializeCommentReplyUI(contentItemId, activeCommentId)InitializeCommentReplyUI(contentItemId, activeCommentId)InitializeCommentReplyUI(contentItemId, activeCommentId)&lt;/p&gt;" />
            <CommentsPart CommentsShown="true" CommentsActive="true" ThreadedComments="true" />
            <CommonPart Owner="/User.UserName=admin" Container="/alias=" CreatedUtc="2016-10-16T17:45:00Z" PublishedUtc="2016-10-16T17:45:55.5446975Z" ModifiedUtc="2016-10-16T17:45:55.5386935Z" />
            <TagsPart Tags="" />
            <TitlePart Title="third post" />
          </BlogPost>
          <Comment Id="/Identifier=023fba3d6e3c49c788d49e7c84b2a605" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="9.43750" Status="Approved" CommentDateUtc="2016-10-16T11:06:09Z" CommentText="Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui. Proin eget tortor risus. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Proin eget tortor risus. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Donec rutrum congue leo eget malesuada. Nulla quis lorem ut libero malesuada feugiat. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui." CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" RepliedOn="/Identifier=91cbba33924c4e10aebf7612af162783" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T11:06:09.8664479Z" PublishedUtc="2016-10-16T11:06:09.8869623Z" ModifiedUtc="2016-10-16T11:06:09.8759542Z" />
            <IdentityPart Identifier="023fba3d6e3c49c788d49e7c84b2a605" />
          </Comment>
          <Comment Id="/Identifier=03ea56784c4b4214ba3fb56dec0d70d7" Status="Published">
            <CommentPart Author="admin" UserName="admin" Email="matthew@example.com" Position="42.00000" Status="Approved" CommentDateUtc="2016-10-17T21:39:49Z" CommentText="another" CommentedOn="/alias=single-thread-jed" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-17T21:39:49.6049524Z" PublishedUtc="2016-10-17T21:39:49.6239662Z" ModifiedUtc="2016-10-17T21:39:49.6124575Z" />
            <IdentityPart Identifier="03ea56784c4b4214ba3fb56dec0d70d7" />
          </Comment>
          <Comment Id="/Identifier=06f61c4fce024be4bc4d7f2cf9633a30" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="34.00000" Status="Approved" CommentDateUtc="2016-10-16T20:22:28Z" CommentText="testing" CommentedOn="/alias=third-post" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T20:22:28.0429558Z" PublishedUtc="2016-10-16T20:22:28.0619693Z" ModifiedUtc="2016-10-16T20:22:28.0499608Z" />
            <IdentityPart Identifier="06f61c4fce024be4bc4d7f2cf9633a30" />
          </Comment>
          <Comment Id="/Identifier=0ead90353d914917b62cc5b77ff16579" Status="Published">
            <CommentPart Author="admin" UserName="admin" Email="matthew@example.com" Position="43.00000" Status="Approved" CommentDateUtc="2016-10-17T21:39:53Z" CommentText="moar" CommentedOn="/alias=single-thread-jed" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-17T21:39:53.026383Z" PublishedUtc="2016-10-17T21:39:53.0488988Z" ModifiedUtc="2016-10-17T21:39:53.0348892Z" />
            <IdentityPart Identifier="0ead90353d914917b62cc5b77ff16579" />
          </Comment>
          <Comment Id="/Identifier=1017c83324ab4540979703ac60ac3230" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="18.87500" Status="Approved" CommentDateUtc="2016-10-16T11:06:04Z" CommentText="Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui. Proin eget tortor risus. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Proin eget tortor risus. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Donec rutrum congue leo eget malesuada. Nulla quis lorem ut libero malesuada feugiat. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui." CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" RepliedOn="/Identifier=38065199d380419ebcf790f5bb74d45b" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T11:06:04.6787623Z" PublishedUtc="2016-10-16T11:06:04.696775Z" ModifiedUtc="2016-10-16T11:06:04.6862677Z" />
            <IdentityPart Identifier="1017c83324ab4540979703ac60ac3230" />
          </Comment>
          <Comment Id="/Identifier=2a41e364ca6d44d38e2c54320dca153f" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="22.00000" Status="Approved" CommentDateUtc="2016-10-16T11:03:54Z" CommentText="&lt;script&gt;alert(&quot;hello&quot;)&lt;/script&gt;" CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T11:03:54.3891831Z" PublishedUtc="2016-10-16T11:03:54.4081966Z" ModifiedUtc="2016-10-16T11:03:54.3966886Z" />
            <IdentityPart Identifier="2a41e364ca6d44d38e2c54320dca153f" />
          </Comment>
          <Comment Id="/Identifier=38065199d380419ebcf790f5bb74d45b" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="16.00000" Status="Approved" CommentDateUtc="2016-10-15T22:32:07Z" CommentText="testing" CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-15T22:32:07.1497895Z" PublishedUtc="2016-10-15T22:32:07.8878136Z" ModifiedUtc="2016-10-15T22:32:07.8107589Z" />
            <IdentityPart Identifier="38065199d380419ebcf790f5bb74d45b" />
          </Comment>
          <Comment Id="/Identifier=4602910fe8de41339fac8612326789af" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="18.50000" Status="Approved" CommentDateUtc="2016-10-16T11:05:53Z" CommentText="s. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Proin eget tortor risus. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Donec rutrum congue leo eget malesuada. Nulla quis lorem ut libero malesuada feugiat. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui." CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" RepliedOn="/Identifier=38065199d380419ebcf790f5bb74d45b" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T11:05:53.1135384Z" PublishedUtc="2016-10-16T11:05:53.1315512Z" ModifiedUtc="2016-10-16T11:05:53.1210436Z" />
            <IdentityPart Identifier="4602910fe8de41339fac8612326789af" />
          </Comment>
          <Comment Id="/Identifier=5e7e7efacbf440a79dae4d6d3ad85a28" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="20.00000" Status="Approved" CommentDateUtc="2016-10-16T09:18:16Z" CommentText="Hello different post to main tester" CommentedOn="/alias=example-with-featured-image" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T09:18:16.116277Z" PublishedUtc="2016-10-16T09:18:16.1352905Z" ModifiedUtc="2016-10-16T09:18:16.1237822Z" />
            <IdentityPart Identifier="5e7e7efacbf440a79dae4d6d3ad85a28" />
          </Comment>
          <Comment Id="/Identifier=6f6bb92e30564e189a80178dc4f72503" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="21.00000" Status="Approved" CommentDateUtc="2016-10-16T11:03:18Z" CommentText="&lt;script&gt;" CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T11:03:18.5371944Z" PublishedUtc="2016-10-16T11:03:18.6367647Z" ModifiedUtc="2016-10-16T11:03:18.552705Z" />
            <IdentityPart Identifier="6f6bb92e30564e189a80178dc4f72503" />
          </Comment>
          <Comment Id="/Identifier=7b901fdbb914484abbf00019c0d7c29d" Status="Published">
            <CommentPart Author="admin" UserName="admin" Email="matthew@example.com" Position="22.00000" Status="Approved" CommentDateUtc="2016-10-17T22:29:57Z" CommentText="Im not the default thread am i" CommentedOn="/alias=example-with-featured-image" CommentedOnContainer="/alias=" RepliedOn="/Identifier=92a3eeb86ad34711961bd05e7e087be0" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-17T22:29:57.9286991Z" PublishedUtc="2016-10-17T22:29:57.9837382Z" ModifiedUtc="2016-10-17T22:29:57.9482131Z" />
            <IdentityPart Identifier="7b901fdbb914484abbf00019c0d7c29d" />
          </Comment>
          <Comment Id="/Identifier=8e5ca536f9f44ad6a8eeeb2f8f60292a" Status="Published">
            <CommentPart Author="Matthew Harris" SiteName="http://www.example.com/" Email="matthew@rtpdesign.co.uk" Position="14.50000" Status="Approved" CommentDateUtc="2016-10-16T20:53:56Z" CommentText="jkkjh##&#xD;&#xA;&#xD;&#xA;no id's were harmed in the making of this comment" CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" RepliedOn="/Identifier=2a41e364ca6d44d38e2c54320dca153f" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T20:53:56.3389691Z" PublishedUtc="2016-10-16T20:53:56.358983Z" ModifiedUtc="2016-10-17T10:42:36.9937806Z" />
            <IdentityPart Identifier="8e5ca536f9f44ad6a8eeeb2f8f60292a" />
          </Comment>
          <Comment Id="/Identifier=91cbba33924c4e10aebf7612af162783" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="18.75000" Status="Approved" CommentDateUtc="2016-10-16T11:05:58Z" CommentText="Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui. Proin eget tortor risus. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Proin eget tortor risus. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Donec rutrum congue leo eget malesuada. Nulla quis lorem ut libero malesuada feugiat. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui." CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" RepliedOn="/Identifier=38065199d380419ebcf790f5bb74d45b" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T11:05:58.2236688Z" PublishedUtc="2016-10-16T11:05:58.2441835Z" ModifiedUtc="2016-10-16T11:05:58.2321747Z" />
            <IdentityPart Identifier="91cbba33924c4e10aebf7612af162783" />
          </Comment>
          <Comment Id="/Identifier=92a3eeb86ad34711961bd05e7e087be0" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="21.00000" Status="Approved" CommentDateUtc="2016-10-16T20:11:57Z" CommentText="add comment" CommentedOn="/alias=example-with-featured-image" CommentedOnContainer="/alias=" RepliedOn="/Identifier=5e7e7efacbf440a79dae4d6d3ad85a28" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T20:11:57.6549502Z" PublishedUtc="2016-10-16T20:11:57.722498Z" ModifiedUtc="2016-10-16T20:11:57.6624556Z" />
            <IdentityPart Identifier="92a3eeb86ad34711961bd05e7e087be0" />
          </Comment>
          <Comment Id="/Identifier=9918cbdffae248708c08b6c5d435b4cc" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="10.43750" Status="Approved" CommentDateUtc="2016-10-16T11:06:15Z" CommentText="Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui. Proin eget tortor risus. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Proin eget tortor risus. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Donec rutrum congue leo eget malesuada. Nulla quis lorem ut libero malesuada feugiat. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui." CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" RepliedOn="/Identifier=023fba3d6e3c49c788d49e7c84b2a605" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T11:06:15.8466984Z" PublishedUtc="2016-10-16T11:06:15.8642107Z" ModifiedUtc="2016-10-16T11:06:15.8542036Z" />
            <IdentityPart Identifier="9918cbdffae248708c08b6c5d435b4cc" />
          </Comment>
          <Comment Id="/Identifier=a99ce05c2af24ac8b56a8170ac50824e" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="17.00000" Status="Approved" CommentDateUtc="2016-10-15T22:32:16Z" CommentText="testing" CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" RepliedOn="/Identifier=38065199d380419ebcf790f5bb74d45b" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-15T22:32:16.1556869Z" PublishedUtc="2016-10-15T22:32:16.1742005Z" ModifiedUtc="2016-10-15T22:32:16.1641934Z" />
            <IdentityPart Identifier="a99ce05c2af24ac8b56a8170ac50824e" />
          </Comment>
          <Comment Id="/Identifier=c48f2ad58bd548e5a46c529e1500363b" Status="Published">
            <CommentPart Author="admin" UserName="admin" Email="matthew@example.com" Position="45.00000" Status="Approved" CommentDateUtc="2016-10-17T21:40:15Z" CommentText="from main thread" CommentedOn="/alias=single-thread-jed" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-17T21:40:15.7405319Z" PublishedUtc="2016-10-17T21:40:15.7605461Z" ModifiedUtc="2016-10-17T21:40:15.7490378Z" />
            <IdentityPart Identifier="c48f2ad58bd548e5a46c529e1500363b" />
          </Comment>
          <Comment Id="/Identifier=c4cf970b235249389d2960b3016d79a1" Status="Published">
            <CommentPart Author="admin" UserName="admin" Email="matthew@example.com" Position="47.00000" Status="Approved" CommentDateUtc="2016-10-17T22:30:32Z" CommentText="but i am a root comment" CommentedOn="/alias=example-with-featured-image" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-17T22:30:32.1420045Z" PublishedUtc="2016-10-17T22:30:32.161018Z" ModifiedUtc="2016-10-17T22:30:32.1495099Z" />
            <IdentityPart Identifier="c4cf970b235249389d2960b3016d79a1" />
          </Comment>
          <Comment Id="/Identifier=dfdb7dbb0eb84638b90585192798b23a" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="9.37500" Status="Approved" CommentDateUtc="2016-10-16T20:18:40Z" CommentText=" sit amet nisl tempus convallis quis ac lectus. Donec rutrum congue leo eget malesuada. Nulla quis lorem ut libero malesuada feugiat. Vestibulum ac diam sit" CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" RepliedOn="/Identifier=4602910fe8de41339fac8612326789af" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T20:18:40.2911225Z" PublishedUtc="2016-10-16T20:18:40.3106365Z" ModifiedUtc="2016-10-16T20:18:40.2991282Z" />
            <IdentityPart Identifier="dfdb7dbb0eb84638b90585192798b23a" />
          </Comment>
          <Comment Id="/Identifier=f43b14342e6a40ee97387c78e3ce3498" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="36.00000" Status="Approved" CommentDateUtc="2016-10-16T21:51:22Z" CommentText="testing root comment" CommentedOn="/alias=third-post" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T21:51:22.4140031Z" PublishedUtc="2016-10-16T21:51:22.4825517Z" ModifiedUtc="2016-10-16T21:53:01.3047967Z" />
            <IdentityPart Identifier="f43b14342e6a40ee97387c78e3ce3498" />
          </Comment>
          <Comment Id="/Identifier=f89d3114508a450b8ca17d9a87e6ea26" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="18.00000" Status="Approved" CommentDateUtc="2016-10-16T11:05:46Z" CommentText="get tortor risus. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Donec rutrum congue leo eget malesuada. Nulla quis lorem ut libero malesuada feugiat. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui." CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" RepliedOn="/Identifier=38065199d380419ebcf790f5bb74d45b" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T11:05:46.7460152Z" PublishedUtc="2016-10-16T11:05:46.7645279Z" ModifiedUtc="2016-10-16T11:05:46.7545207Z" />
            <IdentityPart Identifier="f89d3114508a450b8ca17d9a87e6ea26" />
          </Comment>
          <Comment Id="/Identifier=fa65f912ffc440fd9802c4cfff77e9e5" Status="Published">
            <CommentPart Author="admin" UserName="admin" Email="matthew@example.com" Position="44.00000" Status="Approved" CommentDateUtc="2016-10-17T21:39:57Z" CommentText="AGAIN!" CommentedOn="/alias=single-thread-jed" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-17T21:39:57.6391601Z" PublishedUtc="2016-10-17T21:39:57.6596751Z" ModifiedUtc="2016-10-17T21:39:57.6476663Z" />
            <IdentityPart Identifier="fa65f912ffc440fd9802c4cfff77e9e5" />
          </Comment>
          <Comment Id="/Identifier=fafae808d2db42d5879c0762fb3e2ef0" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="19.00000" Status="Approved" CommentDateUtc="2016-10-16T09:17:23Z" CommentText="New comment new thread" CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T09:17:22.9760105Z" PublishedUtc="2016-10-16T09:17:23.1801549Z" ModifiedUtc="2016-10-16T09:17:23.0990977Z" />
            <IdentityPart Identifier="fafae808d2db42d5879c0762fb3e2ef0" />
          </Comment>
          <Comment Id="/Identifier=fe6bf6f65c97400d90bb1de39553a31a" Status="Published">
            <CommentPart Author="admin" UserName="admin" Position="29.00000" Status="Approved" CommentDateUtc="2016-10-16T11:06:56Z" CommentText="Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui. Proin eget tortor risus. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Proin eget tortor risus. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Donec rutrum congue leo eget malesuada. Nulla quis lorem ut libero malesuada feugiat. Vestibulum ac diam sit amet quam vehicula elementum sed sit amet dui." CommentedOn="/alias=test-post-1" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-16T11:06:56.6121714Z" PublishedUtc="2016-10-16T11:06:56.6326861Z" ModifiedUtc="2016-10-16T11:06:56.6211778Z" />
            <IdentityPart Identifier="fe6bf6f65c97400d90bb1de39553a31a" />
          </Comment>
          <Comment Id="/Identifier=ff868141d8704b28acac42321e42c17e" Status="Published">
            <CommentPart Author="admin" UserName="admin" Email="matthew@example.com" Position="41.00000" Status="Approved" CommentDateUtc="2016-10-17T21:39:44Z" CommentText="testing the single thread" CommentedOn="/alias=single-thread-jed" CommentedOnContainer="/alias=" />
            <CommonPart Owner="/User.UserName=admin" CreatedUtc="2016-10-17T21:39:44.736994Z" PublishedUtc="2016-10-17T21:39:44.7580098Z" ModifiedUtc="2016-10-17T21:39:44.745Z" />
            <IdentityPart Identifier="ff868141d8704b28acac42321e42c17e" />
          </Comment>
        </Content>
      </Orchard>

File: src/Orchard.Web/Modules/Orchard.Comments/Views/EditorTemplates/Parts.Comment.cshtml [hunk 1]
```diff
@@ -1,38 +1,37 @@
...
-
 @if (WorkContext.CurrentUser == null) {
     <fieldset class="who">
-        <legend id="add-comment">@T("Add a Comment")</legend>
+        <legend class="add-comment">@T("Add a Comment")</legend>
         <ol>
             <li>
-                @Html.LabelFor(m => m.Author, T("Name"))
-                @Html.TextBoxFor(m => m.Author)
+                @Html.LabelFor(m => m.Author, T("Name").Text, new { id = string.Format("Comments_Author_{0}", Model.CommentedOn) })
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Comments/Views/EditorTemplates/Parts.Comment.cshtml]: Use <label>, and everywhere there is a LabelFor/TextboxFor

**Comment by sebastienros** [comment]
We forgot to merge it. Matthew please do it if you are done with it.

**Pull request merged**